### PR TITLE
Feature/cdap 15632 nosql impl query parser

### PR DESCRIPTION
**Pull request by vjusiega**

### CODE REVIEW

File: cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java [hunk 1]
```diff
@@ -304,22 +312,76 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, SearchRequest r
...
+      else{
+        int count = hashedResults.get(entity).get(label);
+        hashedResults.get(entity).put(label, count + 1);
+      }
+    }
+    return hashedResults;
+  }
+
+  private Map<MetadataEntity, Map<String, Integer>> filterEntries(Map<MetadataEntity, Map<String, Integer>> results,
+                                                         SearchRequest request) {
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java:337]: nit: alignment

File: cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java [hunk 2]
```diff
@@ -304,22 +312,76 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, SearchRequest r
...
+      }
+      else{
+        int count = hashedResults.get(entity).get(label);
+        hashedResults.get(entity).put(label, count + 1);
+      }
+    }
+    return hashedResults;
+  }
+
+  private Map<MetadataEntity, Map<String, Integer>> filterEntries(Map<MetadataEntity, Map<String, Integer>> results,
```
Comment (CuriousVini) [inline cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java:336]: please add a javadoc on what filterEntries class is doing.

File: cdap-metadata-spi/src/test/java/io/cdap/cdap/spi/metadata/MetadataStorageTest.java [hunk 3]
```diff
@@ -1509,6 +1509,7 @@ private void testSortedSearch(MetadataStorage mds, List<MetadataRecord> records,
   private void testSortedSearch(MetadataStorage mds, Sorting sorting, List<MetadataRecord> sorted) throws IOException {
     // search once with sort by creation time and validate that records are in order
     SearchResponse response = mds.search(SearchRequest.of("*").setSorting(sorting).build());
+    response.getResults();
```
Comment (CuriousVini) [inline cdap-metadata-spi/src/test/java/io/cdap/cdap/spi/metadata/MetadataStorageTest.java:1512]: nit: unintentional change?

**Review by CuriousVini — COMMENTED** [review]
also add unit tests to search required terms.

File: cdap-metadata-spi/src/test/java/io/cdap/cdap/spi/metadata/MetadataStorageTest.java (hunk 3 above)
Comment (vjusiega) [reply cdap-metadata-spi/src/test/java/io/cdap/cdap/spi/metadata/MetadataStorageTest.java:1512]: yes, thank you

**Pull request closed without merging**